capability.go

```diff
@@ -472,32 +494,48 @@ func (c Client) IsValid() bool {
 // are not fully resolved: use Resolve if this is an issue.  If either
 // c or c2 are released, then IsSame panics.
 func (c Client) IsSame(c2 Client) bool {
-	h1, _, released := c.peek()
+	h1, _, released := c.startCall()
+	defer h1.Release()
 	if released {
 		panic("IsSame on released client")
 	}
-	h2, _, released := c2.peek()
+	h2, _, released := c2.startCall()
+	defer h2.Release()
 	if released {
 		panic("IsSame on released client")
 	}
-	return h1 == h2
+	valid1 := h1.IsValid()
+	valid2 := h2.IsValid()
+	if !valid1 && !valid2 {
+		return true
+	}
+	if !valid1 || !valid2 {
+		return false
+	}
+	return h1.Value() == h2.Value()
 }
 
 // Resolve blocks until the capability is fully resolved or the Context is Done.
 // Resolve only returns an error if the context is canceled; it returns nil even
 // if the capability resolves to an error.
 func (c Client) Resolve(ctx context.Context) error {
 	for {
-		h, resolved, released := c.peek()
+		h, _, released := c.startCall()
+		defer h.Release()
 		if released {
 			return errors.New("cannot resolve released client")
 		}
 
-		if resolved {
+		if h == nil {
```

Comment:
Is this right?  I'd expect a nil `clientHook` to _not_ be resolved.

Reply:
I can't remember why I changed this. Pushed another commit that changes it back.